Sources/AccioKit/Services/GitResetService.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+final class GitResetService {
+    static let shared = GitResetService()
+
+    private init() { }
+
+    func resetGit(atPath directory: String) throws {
+        try bash("git -C '\(directory)' reset HEAD --hard --quiet 2> /dev/null")
+        try bash("git -C '\(directory)' clean -fd --quiet 2> /dev/null")
```

Comment:
Why is the `-fdX` variant missing here?

Reply:
I think we should add a parameter to `resetGit` named something like `includeUntrackedFiles: Bool` and run the `-fdX` variant in that case, too.

Reply:
For me the documentation I read sounded as calling the `-fdX` again after having called `-fd` was redundant, as `-fd` already did everything. Did I get that wrong?

Reply:
Yes, you did get this wrong if I recall correctly. In my tests calling `-fd` wasn't enough to also reset files that were ignored by entries in the `.gitignore` files which could lead to problems with SwiftPM only projects. The `-rfX` call solved the issue via https://github.com/JamitLabs/Accio/commit/94323c14b16c4450990db2fcf495e8d31df37523. I don't think I made the commit just in case, I'm pretty sure it solved a real problem.

Reply:
Okay, I'll adjust accordingly. Is there a special reason `-fdX` wasn't previously used everywhere? If not, we could just add it without making it configurable via a param.

Reply:
I'm not sure, it might only be that I wanted to save time because it was not needed in some situations (and the reset could take some time). So I'd rather keep the parameter, but feel free to default it to `true`.

Reply:
Please make sure to name it with something better than `fdX`, for example `includeUntrackedFiles` as suggested above.